charts/datadog/templates/cluster-agent-deployment.yaml

```diff
@@ -178,14 +178,24 @@ spec:
             value: {{ .Values.clusterAgent.admissionController.mutateUnlabelled | quote }}
           - name: DD_ADMISSION_CONTROLLER_SERVICE_NAME
             value: {{ template "datadog.fullname" . }}-cluster-agent-admission-controller
-          {{- if .Values.clusterAgent.admissionController.configMode }}
           - name: DD_ADMISSION_CONTROLLER_INJECT_CONFIG_MODE
-            value: {{ .Values.clusterAgent.admissionController.configMode | quote }}
-          {{- if eq .Values.clusterAgent.admissionController.configMode "service" }}
+            {{- if .Values.clusterAgent.admissionController.configMode }}
+            value: {{ .Values.clusterAgent.admissionController.configMode }}
+            {{- else if eq (include "trace-agent-use-tcp-port" .) "true" }}
+            {{- if eq (include "enable-service-internal-traffic-policy" .) "true" }}
+            value: service
+            {{- else }}
+            value: hostip
+            {{- end }}
+            {{- else if eq (include "trace-agent-use-uds" .) "true" }}
+            value: socket
+            {{- else }}
+            value: socket
+            {{- end }}
+          {{- if or (eq .Values.clusterAgent.admissionController.configMode "service") (and (eq (include "trace-agent-use-tcp-port" .) "true") (eq (include "enable-service-internal-traffic-policy" .) "true")) }}
```

Comment:
This condition is long and difficult to read, it's completely fine to remove it and always set `DD_ADMISSION_CONTROLLER_INJECT_CONFIG_LOCAL_SERVICE_NAME` regardless of the mode.